pkg/networkaware/networkoverhead/networkoverhead.go

```diff
@@ -0,0 +1,577 @@
+
+				if no.weightsName != ntv1alpha1.NetworkTopologyNetperfCosts {
+					// Sort Costs by origin, might not be sorted since were manually defined
+					sort.Sort(networkawareutil.ByOrigin(topologyList))
+				}
+
+				// Binary search through TopologyList: find the costs for the given Region
+				costs := networkawareutil.FindOriginCosts(topologyList, zone)
+
+				// Add Zone Costs
+				for _, c := range costs {
+					costMap[networkawareutil.CostKey{ // Add the cost to the map
+						Origin:      zone,
+						Destination: c.Destination}] = c.NetworkCost
+				}
+			}
+			seen = true
+		} else if seen == true { // Costs are sorted by origin, thus stop here
+			break
+		}
+	}
+}
+
+// checkMaxNetworkCostRequirements : verifies the number of met and unmet dependencies based on the pod being filtered
+func checkMaxNetworkCostRequirements(scheduledList networkawareutil.ScheduledList, dependencyList []agv1alpha1.DependenciesInfo, nodeInfo *framework.NodeInfo, region string,
+	zone string, costMap map[networkawareutil.CostKey]int64, no *NetworkOverhead) (int64, int64, error) {
+
+	var numOK int64 = 0
+	var numNotOK int64 = 0
+
+	// check if maxNetworkCost fits
+	for _, podAllocated := range scheduledList { // For each pod already allocated
+		if podAllocated.Hostname != "" { // if hostname not empty...
+			for _, d := range dependencyList { // For each pod dependency
+				if podAllocated.Selector == d.Workload.Selector { // If the pod allocated is an established dependency
+					if podAllocated.Hostname == nodeInfo.Node().Name { // If the Pod hostname is the node being filtered, requirements are checked via extended resources
+						numOK += 1
+					} else { // If Nodes are not the same
+						// Get NodeInfo from pod Hostname
+						podHostname, err := no.handle.SnapshotSharedLister().NodeInfos().Get(podAllocated.Hostname)
+						if err != nil {
+							klog.ErrorS(nil, "getting pod hostname %q from Snapshot: %v", podHostname, err)
+							return numOK, numNotOK, err
+						}
+
+						// Get zone and region from Pod Hostname
+						regionPodHostname := networkawareutil.GetNodeRegion(podHostname.Node())
+						zonePodHostname := networkawareutil.GetNodeZone(podHostname.Node())
+
+						if regionPodHostname == "" && zonePodHostname == "" { // Node has no zone and region defined
+							numNotOK += 1
+						} else if region == regionPodHostname { // If Nodes belong to the same region
+							if zone == zonePodHostname { // If Nodes belong to the same zone
+								numOK += 1
+							} else { // belong to a different zone, check maxNetworkCost
+								cost, costOK := costMap[networkawareutil.CostKey{ // Retrieve the cost from the map (origin: zone, destination: pod zoneHostname)
+									Origin:      zone, // Time Complexity: O(1)
+									Destination: zonePodHostname,
+								}]
+								if costOK {
+									if cost <= d.MaxNetworkCost {
+										numOK += 1
+									} else {
+										numNotOK += 1
```

Comment:
in this case, there is some constraint violated, instead of comparing the number of the violations with some other number, shouldn't we return to fail immediately?